libkbfs/block_journal.go

```diff
@@ -72,18 +59,20 @@ type blockJournal struct {
 	log      logger.Logger
 	deferLog logger.Logger
 
-	j    diskJournal
-	refs map[BlockID]blockRefMap
+	// j is the main journal.
+	j diskJournal
 
+	// saveUntilMDFlush, when non-nil, prevents garbage collection
+	// of blocks. When removed, all the referenced blocks are
+	// garbage-collected.
+	//
+	// TODO: We only really need to save a list of IDs, and not a
```

Comment:
Thinking about it more, should we be appending to both saveUntilMDFlush and j if the former is non-nil? For example, say revs 1-5 get saved. But then we end up flushing revs 1-10, maybe because the MD server is down or something, or because we have the freedom to always flush block ops first. Then when the MD flush does happen, blocks from 6-10 won't get GCed properly. Am I missing something?